src/main/scala/monocly/Optic.scala

```diff
@@ -0,0 +1,94 @@
+package monocly
+
+import monocly.impl._
+import monocly.functions.{Each, Index}
+import monocly.internal.{NonEmptyList, Applicative, Monoid}
+
+
+sealed trait OpticCan
+
+//        GetMany
+//          /\
+//         /  \
+// GetOption  GetOneOrMore
+//         \  /
+//          \/
+//        GetOne
+trait GetMany extends OpticCan
+trait GetOption extends GetMany
+trait GetOneOrMore extends GetMany
+trait GetOne extends GetOption with GetOneOrMore 
+
+//    Modify
+//      ^
+//      |
+//  ReverseGet
+trait Modify extends OpticCan
+trait ReverseGet extends Modify
+
+type Optic[+ThisCan <: OpticCan, S, A] = POptic[ThisCan, S, S, A, A]
+
+
+final class POptic[+ThisCan <: OpticCan, -S, +T, +A, -B] private[monocly](
```

Comment:
What's the benefit of constraining `ThisCan` to be a subtype of `OpticCan`?

Reply:
Can an external user create an optic with their own `ThisCan` parameter? I though that the constructor was private and  we only offered smart constructors with hardcoded `ThisCan` type.

It is probably just me, but when I read `A <: Foo` or `A: Foo` I assume that the code below relies on this constraint, e.g. for a pattern match or to access typeclass method.